alacritty_config_derive/src/de_struct.rs

```diff
@@ -0,0 +1,218 @@
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{format_ident, quote};
+use syn::parse::{self, Parse, ParseStream};
+use syn::punctuated::Punctuated;
+use syn::spanned::Spanned;
+use syn::{Error, Field, GenericParam, Generics, Ident, LitStr, Token, Type, TypeParam};
+
+/// Error message when attempting to flatten multiple fields.
+const MULTIPLE_FLATTEN_ERROR: &str = "At most one instance of #[config(flatten)] is supported";
+/// Use this crate's name as log target.
+const LOG_TARGET: &str = env!("CARGO_PKG_NAME");
```

Comment:
If you have concerns wrt using target log name, etc in proc macros. You may look into wayland-scanner magic, since it works by assuming that you 'use' certain things, and it'll generate either server or client code for that.

The wayland-scanner usage of quote is a bit complicated, but you have an option to provide custom target if that's what you want.

Since you had a problem of exporting log target info, but you can actually do that the other way around and bring it in(not sure if you can do that conditionally though).

Reply:
I'm not sure what you mean? Import the target from the outside?

That's something I cannot do because it is completely opaque to the user of the library and there's no way to "detect" the imports from the proc macro.

Reply:
I got confused by different rust macros and how they work. Nothing will help from that I've mentioned.